src/program.cpp

```diff
@@ -221,15 +233,42 @@ std::vector<argument> generic_eval(const module& p,
                     assert(results.find(i) != results.end());
                     return results[i];
                 });
-            results.emplace(ins, trace(ins, [&] {
-                                return ins->normalized_operator().compute(
-                                    ctx, ins->get_shape(), values);
-                            }));
+
+            const auto& mod_args = ins->module_inputs();
+            if(not mod_args.empty())
+                results.emplace(
+                    ins, trace(ins, mod, [&] {
+                        return ins->get_operator().compute(
+                            values,
+                            mod_args,
+                            [&](module_ref smod,
+                                const std::unordered_map<std::string, argument>& inputs) {
+                                // wrap up parameters for sub_modules
+                                const auto& param_names = smod->get_parameter_names();
+                                parameter_map m;
+                                for(const auto& nm : param_names)
+                                    if(contains(inputs, nm))
+                                        m[nm] = inputs.at(nm);
+                                    else if(contains(params, nm))
+                                        m[nm] = params.at(nm);
+                                    else
+                                        MIGRAPHX_THROW("Input " + nm + " parameter of module: \"" +
+                                                       mod->name() + "\" not exist!");
+                                return generic_eval(smod, ctx, m, results, trace);
+                            });
+                    }));
+            else
+            {
+                results.emplace(ins, trace(ins, mod, [&] {
+                                    return ins->normalized_operator().compute(
```

Comment:
Is this necessary? Won't `compute` call the non-module overload when the modules are empty? If not we can change this in a future PR, and no need to change here.

Reply:
Currently, the compute function with mod_args input does not have the context input. As I talked to you before, if context is added as an input, I met the problem of the compute_function not found in gpu computation.